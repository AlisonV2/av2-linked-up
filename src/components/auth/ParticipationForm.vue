<template>
  <form class="auth-form" @submit.prevent="setParticipation">
    <div class="row">
      <div class="col-12 text-center mb-4">
        <h2 class="title">Submit participation</h2>
      </div>
    </div>
    <div class="form-floating mb-3">
      <input
        type="number"
        class="form-control"
        placeholder="Please enter an amount of tickets"
        v-model="tickets"
        required
      />
      <label>How many tickets do you want ?</label>
    </div>
    <div class="error mb-3 mt-3">{{ error }}</div>
    <div class="btn-center">
      <app-button>Confirm</app-button>
    </div>
    <slot />
  </form>
</template>

<script>
import * as Sentry from '@sentry/vue';

/**
 * @exports ParticipationForm
 * @type {Component}
 * @requires Sentry
 * @vue-data {number} tickets - v-model
 * @vue-event setParticipation
 */
export default {
  props: ['id'],
  data() {
    return {
      tickets: null,
      error: null,
    };
  },
  methods: {
    setParticipation() {
      const data = {
        eventId: this.id,
        tickets: this.tickets,
      };
      this.$store
        .dispatch('setParticipation', data)
        // .then(() => {
        //   this.$router.push({ name: 'AdminEvents' });
        // })
        .catch((err) => {
          this.error = err.message;
          Sentry.captureException(err);
          return;
        });
    },
  },
};
</script>

<style lang="scss" scoped>
.title {
  color: $dark;
  font-size: 1.8rem;
  margin-bottom: 1.5rem;
}
</style>
