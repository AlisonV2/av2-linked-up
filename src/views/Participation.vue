<template>
  <div class="container col-6 home">
    <div>
      <ParticipationForm :id="id"/>
    </div>
  </div>
</template>

<script>
import ParticipationForm from '@/components/auth/ParticipationForm';

/**
 * @exports ParticipationPassword
 * @requires ParticipationForm
 * @type {Page}
 */
export default {
  name: 'ParticipationPassword',
  components: {
    ParticipationForm,
  },
  computed: {
    id() {
      return this.$route.params.id;
    }
  }
};
</script>
