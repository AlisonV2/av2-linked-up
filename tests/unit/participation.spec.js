import { shallowMount } from '@vue/test-utils';
import Participation from '@/views/Participation.vue';
import ParticipationForm from '@/components/auth/ParticipationForm';

const $route = {
  params: { id: '1' },
};

const config = {
  data() {
    return {
      id: '1',
    };
  },
  global: {
    components: {
      ParticipationForm,
    },
    mocks: {
      $route,
    },
  },
};
/**
 * @module ParticipationPageTest
 */
describe('Participation.vue', () => {
  it('ParticipationPage Snapshot', () => {
    const component = shallowMount(Participation, config);
    expect(component.element).toMatchSnapshot();
  });

  it('Check if ParticipationForm is rendered', () => {
    const component = shallowMount(Participation, config);
    const form = component.findComponent(ParticipationForm);
    expect(form.exists()).toBe(true);
  });

  it('Check if ParticipationForm is rendered', () => {
    const component = shallowMount(Participation, config);
    const computed = component.vm.id;
    expect(computed).toBe('1');
  });
});
