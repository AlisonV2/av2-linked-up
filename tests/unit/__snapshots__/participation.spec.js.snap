// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participation.vue ParticipationPage Snapshot 1`] = `
<div
  class="container col-6 home"
>
  <div>
    <participation-form-stub
      id="1"
    />
  </div>
</div>
`;
